.github/workflows/test.yml

```diff
@@ -1,6 +1,12 @@
 name: Unit tests
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
```

Comment:
Why do we only perform the unit tests on the `dev`, `main` and `master` branches? I feel like we want to keep this on every branch?

Reply:
This is because we have redundant and unnecessary tests in the PRs (push AND pull_request). But we only need to track changes in PRs (and not in branches that aren't linked to a PR).
Concerning the `dev` branch, it is intended to manage better versions in the future.